tests/scripts/generate_test_code.py

```diff
@@ -0,0 +1,722 @@
+#!/usr/bin/env python3
+# Test suites code generator.
+#
+# Copyright (C) 2018, ARM Limited, All Rights Reserved
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This file is part of mbed TLS (https://tls.mbed.org)
+
+"""
+Test Suite code generator.
+
+Generates a test source file using following input files:
+
+test_suite_xyz.function - Read test functions from test suite functions file.
+test_suite_xyz.data - Read test functions and their dependencies to generate
+                      dispatch and dependency check code.
+main template - Substitute generated test function dispatch code, dependency
+                checking code.
+platform .function - Read host or target platform implementation for
+                     dispatching test cases from .data file.
+helper .function - Read common reusable functions.
+"""
+
+
+import io
+import os
+import re
+import sys
+import argparse
+import shutil
+
+
+BEGIN_HEADER_REGEX = '/\*\s*BEGIN_HEADER\s*\*/'
+END_HEADER_REGEX = '/\*\s*END_HEADER\s*\*/'
+
+BEGIN_SUITE_HELPERS_REGEX = '/\*\s*BEGIN_SUITE_HELPERS\s*\*/'
+END_SUITE_HELPERS_REGEX = '/\*\s*END_SUITE_HELPERS\s*\*/'
+
+BEGIN_DEP_REGEX = 'BEGIN_DEPENDENCIES'
+END_DEP_REGEX = 'END_DEPENDENCIES'
+
+BEGIN_CASE_REGEX = '/\*\s*BEGIN_CASE\s*(.*?)\s*\*/'
+END_CASE_REGEX = '/\*\s*END_CASE\s*\*/'
+
+
+class InvalidFileFormat(Exception):
+    """
+    Exception to indicate invalid file format.
+    """
+    pass
+
+
+class FileWrapper(io.FileIO):
+    """
+    File wrapper class. Provides reading with line no. tracking.
+    """
+
+    def __init__(self, file_name):
+        """
+        Init file handle.
+
+        :param file_name: File path to open.
+        """
+        super(FileWrapper, self).__init__(file_name, 'r')
+        self.line_no = 0
+
+    def __next__(self):
```

Comment:
The method is called `next`, not `__next__`. Here you're overriding a method name that's never called. As a consequence, all generated code has `#line 1 "…"` because `line_no` never gets incremented.

Reply:
Does the assignment at line https://github.com/ARMmbed/mbedtls/pull/930/files/9c169994a6630948d7f60223d6b5e08320fa6f07#diff-5801853c8d79838f04882c360fe720b3R103 not work?

Reply:
I tested it on Ubuntu with both Python 2 & 3 and it works fine. Though with Pylint I found that `FileIO` class had `next` and not `__next__`. So I have swapped the names in coming commit.

Reply:
When I wrote this comment, I'd used the script, using the version of the language implied by the shebang, and I noticed that all line numbers were coming out as 1.

Your new commit is the patch I suggested, isn't it? That should work.

Reply:
Ah sorry I didn't realise it was an old comment, since it still showed up in the summary page.